packages/mongodb-rag-ingest/src/embed/updateEmbeddedContent.ts

```diff
@@ -127,25 +136,25 @@ export const updateEmbeddedContentForPage = async ({
 
   const embeddedContent: EmbeddedContent[] = [];
 
-  // Process sequentially because we're likely to hit rate limits before any
-  // other performance bottleneck
-  for (let i = 0; i < contentChunks.length; ++i) {
-    const chunk = contentChunks[i];
-    logger.info(
-      `Vectorizing chunk ${i + 1}/${contentChunks.length} for ${
-        page.sourceName
-      }: ${page.url}`
-    );
-    const { embedding } = await embedder.embed({
-      text: chunk.text,
-    });
-    embeddedContent.push({
-      ...chunk,
-      embedding,
-      updated: new Date(),
-      chunkAlgoHash,
-    });
-  }
+  await PromisePool
+    .withConcurrency(concurrencyOptions?.embed?.createChunks || 1)
+    .for(contentChunks)
+    .process(async (chunk, index, pool) => {
+      logger.info(
+        `Vectorizing chunk ${index + 1}/${contentChunks.length} for ${
+          page.sourceName
+        }: ${page.url}`
+      );
+      const { embedding } = await embedder.embed({
+        text: chunk.text,
+      });
+      embeddedContent.push({
+        ...chunk,
+        embedding,
+        updated: new Date(),
+        chunkAlgoHash,
+      });
+    })
```

Comment:
rather than pushing into the `embeddedContent` array, you can just return the results from the PromisePool. yields cleaner code and takes advantage of any optimizations in the PromisePool library.

something like:

```ts
const embeddedContent = await PromisePool...process((args)=>{
  // ...
  return {
    // the embedded content object
  }
})
```